### ISSUE TITLE
Number of samples in train set: 147077 

### ISSUE THREAD

**Issue by popov1212** [Rex-sys-hk/PlanScope#5]
Thanks for your great work!

I have only one x.db file under the xxx/trainval folder. However, when I run the train_scope.sh script, the output is:

Number of samples in train set: 147077  
Number of samples in validation set: 29607

I tried setting scenario_filter=all_scenarios in the train_scope.sh script, but it didn’t resolve the issue.

I also tried setting scenario_filer.limit_total_scenarios: 10000/1/0.001
But also didn't solve the issue.

How can I ensure the script correctly uses the data from the x.db file? Could you help me identify and resolve this problem?

**Comment by popov1212** [comment]
when I run the train_scope.sh on mini_split
The result is same:
Number of samples in train set: 147077
Number of samples in validation set: 29607

**Comment by popov1212** [comment]
Thanks in advance for your answer！

**Comment by Rex-sys-hk** [comment]
Maybe you should carefully check the environment variables, especially NUPLAN_EXP_ROOT.

**Comment by popov1212** [comment]
Thanks for your reply, 
I add scenario_filter=all_scenarios in train_scope.sh
now I encounter the results：No scenarios were retrieve for training，check the scenario_filter parameter

could you help me to solve it?

**Comment by popov1212** [comment]
I use the xxx.db files in train_vegas_1

**Comment by popov1212** [comment]
I run the feature cache scripts as ReadME says before running the train_scope.sh

do I need to ignore the feature cache step?

**Comment by Rex-sys-hk** [comment]
You don't have to run feature chaching every time before training. I think I can not figure out the problem by given information, would you mind to paste your training script and folder tree of nuplan dataset here for debugging?

**Comment by popov1212** [comment]
Thanks for your reply ！
Now I try to use the total nuplan set to run the train_scope.sh
I first run the feature cache
and then run the train_scope.sh

the results：assert torch.isfinite(q).all() in hierarchical_decoder.py line 205

I tried to print the q.shape, and the results showed
Its value is nan

**Comment by popov1212** [comment]
Below is the feature cache script I run 
export PYTHONPATH=$PYTHONPATH:$(pwd)

python run_training.py \
    py_func=cache +training=train_scope \
    scenario_builder=nuplan \
    cache.cache_path=/home/my_name/nuplan/exp/cache_pluto_1M \
    cache.cleanup_cache=false \
    scenario_filter=training_scenarios_1M \
    worker.threads_per_node=40

**Comment by popov1212** [comment]
Below attached is my train_scope.sh content 

echo "====Start training====" &&

CUDA_VISIBLE_DEVICES="0,1" python run_training.py \
  py_func=train +training=train_scope \
  worker=single_machine_thread_pool worker.max_workers=32 \
  scenario_builder=nuplan \
  scenario.filter=all_scenarios \
  cache.use_cache_without_dataset=true \
  cache.cache_path=/home/my_name/exp/cache_pluto_1M \
  data_loader.params.batch_size=32 data_loader.params.num_workers=32 \
  lr=1e-3 epochs=35 warmup_epochs=3 weight_decay=0.0001 \
  lightning.trainer.params.val_check_interval=0.5 \
  wandb.mode=online wandb.project=nuplan wandb.name=scope \
  data_loader.datamodule.train_fraction=0.2 \
  data_loader.datamodule.val_fraction=0.2 \
  data_loader.datamodule.test_fraction=0.2 \
  model.cat_x=true model.ref_free_traj=true \
  +custom_trainer.use_contrast_loss=false model.use_hidden_proj=false \
  +custom_trainer.mul_ade_loss=[] \
  +custom_trainer.max_horizon=20 \
  +custom_trainer.dynamic_weight=false \
  +model.wtd_with_history=false +custom_trainer.wtd_with_history=false \
  model.recursive_decoder=true +model.multihead_decoder=false \
  model.future_steps=80 \
  +custom_trainer.learning_output='velocity' \
  +custom_trainer.init_weights=[1.0,1.0,1.0,1.0,1.0,1.0] \
  +custom_trainer.wavelet=['cgau1','constant','haar','constant'] \
  &&

  echo "====Training End===="

**Comment by Rex-sys-hk** [comment]
Not A Number(NaN) problem does happen occasionally especially when you are using GPU number/batch size. You can try reducing the learning rate to stabilize the training process.

**Comment by popov1212** [comment]
Thanks for your answer

now when I run the nuplan_mini, the result is:
assert self.val_sanity_progress_bar_id is not None 

My cache script is:
python run_training.py
py_func=cache +training=train_scope
scenario_builder=nuplan_mini
cache.cache_path=/home/my_name/nuplan/exp/sanity_check
cache.cleanup_cache=false
scenario_filter=training_scenarios_tiny
worker=sequential

my train_scope.sh content is:
echo "====Start training====" &&

CUDA_VISIBLE_DEVICES="0,1" python run_training.py
py_func=train +training=train_scope
worker=single_machine_thread_pool worker.max_workers=32
scenario_builder=nuplan_mini
scenario.filter=training_scenarios_tiny
cache.use_cache_without_dataset=true
cache.cache_path=/home/my_name/nuplan/exp/sanity_check
data_loader.params.batch_size=32 data_loader.params.num_workers=32
lr=1e-3 epochs=35 warmup_epochs=3 weight_decay=0.0001
lightning.trainer.params.val_check_interval=0.5
wandb.mode=online wandb.project=nuplan wandb.name=scope
data_loader.datamodule.train_fraction=0.2
data_loader.datamodule.val_fraction=0.2
data_loader.datamodule.test_fraction=0.2
model.cat_x=true model.ref_free_traj=true
+custom_trainer.use_contrast_loss=false model.use_hidden_proj=false
+custom_trainer.mul_ade_loss=[]
+custom_trainer.max_horizon=20
+custom_trainer.dynamic_weight=false
+model.wtd_with_history=false +custom_trainer.wtd_with_history=false
model.recursive_decoder=true +model.multihead_decoder=false
model.future_steps=80
+custom_trainer.learning_output='velocity'
+custom_trainer.init_weights=[1.0,1.0,1.0,1.0,1.0,1.0]
+custom_trainer.wavelet=['cgau1','constant','haar','constant']
&&

echo "====Training End===="

**Comment by Rex-sys-hk** [comment]
> Thanks for your answer
> 
> now when I run the nuplan_mini, the result is: assert self.val_sanity_progress_bar_id is not None
> 
> My cache script is: python run_training.py py_func=cache +training=train_scope scenario_builder=nuplan_mini cache.cache_path=/home/my_name/nuplan/exp/sanity_check cache.cleanup_cache=false scenario_filter=training_scenarios_tiny worker=sequential
> 
> my train_scope.sh content is: echo "====Start training====" &&
> 
> CUDA_VISIBLE_DEVICES="0,1" python run_training.py py_func=train +training=train_scope worker=single_machine_thread_pool worker.max_workers=32 scenario_builder=nuplan_mini scenario.filter=training_scenarios_tiny cache.use_cache_without_dataset=true cache.cache_path=/home/my_name/nuplan/exp/sanity_check data_loader.params.batch_size=32 data_loader.params.num_workers=32 lr=1e-3 epochs=35 warmup_epochs=3 weight_decay=0.0001 lightning.trainer.params.val_check_interval=0.5 wandb.mode=online wandb.project=nuplan wandb.name=scope data_loader.datamodule.train_fraction=0.2 data_loader.datamodule.val_fraction=0.2 data_loader.datamodule.test_fraction=0.2 model.cat_x=true model.ref_free_traj=true +custom_trainer.use_contrast_loss=false model.use_hidden_proj=false +custom_trainer.mul_ade_loss=[] +custom_trainer.max_horizon=20 +custom_trainer.dynamic_weight=false +model.wtd_with_history=false +custom_trainer.wtd_with_history=false model.recursive_decoder=true +model.multihead_decoder=false model.future_steps=80 +custom_trainer.learning_output='velocity' +custom_trainer.init_weights=[1.0,1.0,1.0,1.0,1.0,1.0] +custom_trainer.wavelet=['cgau1','constant','haar','constant'] &&
> 
> echo "====Training End===="


Sorry for never met this problem
> now when I run the nuplan_mini, the result is: assert self.val_sanity_progress_bar_id is not None
There might be repeated instantiations of Sanity Check/Lightning.

**Comment by popov1212** [comment]
Thanks for your reply.
I have already run the train_scope.sh successfully.
Now I got the **.ckpt file. If I want to show the results on the test set, do I need to run the feature cache like blew？

My cache script is:
python run_training.py
py_func=test +training=train_scope
scenario_builder=nuplan
cache.cache_path=/home/my_name/nuplan/exp/sanity_check
cache.cleanup_cache=false
scenario_filter=all_scenarios
worker=sequential

but how could I set the parameter:
 <scenario_builder>
<scenario_filter>

**Comment by popov1212** [comment]
I have already get the result video 
Actually no need to run feature cache before run sim_scope.sh

**Comment by popov1212** [comment]
I use 34 epoch in trainning, but the result video shows some problem.

the video of 12 epoch result performed better than the video of 18 epoch result. Have you ever met this problem? The 12epoch result can turn right smoothly at the cross scenario but then running out of lane bound
The 18epoch result can hardly turn right at the cross scenario. It's actually stopped at the cross scenario.

I only change two parameters in sim_scope.sh
BUILDER=nuplan_mini
FILTER=mini_demo_scenario
That is to say I only use one scenario to test its result

**Comment by popov1212** [comment]
And I also run the results you opened: PlanScope-h10-m6 in the same scenario:
BUILDER=nuplan_mini
FILTER=mini_demo_scenario
The result vedio shows: after ego car turns right, It suddenly stoped in the straight lane while the gt smothly go forward.

Why the ego car stopped suddenly in the straight lane while expert keeps running in const velocity?
I wonder if there is something wrong with my parameter setting?
